crates/typst-layout/src/grid/rowspans.rs

```diff
@@ -185,10 +183,8 @@ impl GridLayouter<'_> {
     /// Checks if a row contains the beginning of one or more rowspan cells.
     /// If so, adds them to the rowspans vector.
     pub fn check_for_rowspans(&mut self, disambiguator: usize, y: usize) {
-        // We will compute the horizontal offset of each rowspan in advance.
-        // For that reason, we must reverse the column order when using RTL.
-        let offsets = points(self.rcols.iter().copied().rev_if(self.is_rtl));
-        for (x, dx) in (0..self.rcols.len()).rev_if(self.is_rtl).zip(offsets) {
+        let offsets = points(self.rcols.iter().copied());
+        for (x, dx) in (0..self.rcols.len()).zip(offsets) {
```

Comment:
I suspect this double reversal was indeed an oversight... The order of `self.rowspans` changes, but this doesn't seem to affect rowspan layout code as it doesn't consider that order (all information is contained in x / dx), so that was a good catch.

Reply:
This change here only works because, the code in `layout_rowspan` was also changed, otherwise the double reversal would have been necessary. Essentially the same transformation as for the other layout code was done here as well, where now the offset is computed from whichever direction the columns are drawn (LTR or RTL) and then the position is then fixed up afterwards for RTL in `layout_rowspan`: 
```rs
        let dx = if self.is_rtl { self.width - (dx + width) } else { dx };
```

Reply:
You're right, I forgot that `points()` accumulates and thus the order of its inputs matters. :+1: